domains/primitives/messenger/src/endpoint.rs

```diff
@@ -64,3 +64,24 @@ pub trait EndpointHandler<MessageId> {
         resp: EndpointResponse,
     ) -> DispatchResult;
 }
+
+/// Trait that can provide info for a given domain.
+/// This trait is implemented by pallet-receipts since it tracks the necessary info
+/// on Core domains in System domain runtime.
+/// For other runtimes, this is simply a no-op.
+pub trait DomainInfo<Number, Hash, StateRoot> {
+    /// Returns the best known number of a given domain.
+    fn domain_best_number(domain_id: DomainId) -> Option<Number>;
+    /// Returns the known state root of a specific block.
+    fn domain_state_root(domain_id: DomainId, number: Number, hash: Hash) -> Option<StateRoot>;
```

Comment:
These two runtime apis seems to always used altogather, probably make them combined into just one, something like `validate_core_domain_message(domain_id: DomainId, domain_block_info: DomainBlockInfo) -> Result<(), XDMError>`?

Reply:
I do see your point but this trait is implemented in runtime. if I move the validation logic there, then validation is more pluggable and messenger cannot garuntee the verification of XDM. I'm not making it pluggable makes sense at the moment. I felt it was easier to keep the trait simple and build the verification in messenger.